src/expr/src/expr/expr_array_ndims.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::array::ListRef;
+use risingwave_common::types::ScalarRefImpl;
+use risingwave_expr_macro::function;
+
+/// Returns the number of dimensions of the array.
+///
+/// ```sql
+/// array_ndims ( array anyarray) → int64
+/// ```
+///
+/// Examples:
+///
+/// ```slt
+/// query T
+/// select array_ndims(null::int[]);
+/// ----
+/// NULL
+///
```

Comment:
```suggestion
///
/// ```slt
/// select array_ndims(array[]::int[]);
/// ----
/// NULL
```
One edge case from PG. Currently this returns 1 in the implementation. (Although I'm not very uncomfortable about the difference..